src/MovieApp.java

```diff
@@ -0,0 +1,211 @@
+        Movie minimum = movies[0];
+        Movie maximum = movies[0];
+        for (Movie movie : movies) {
+            sum += movie.getRating();
+            if (movie.getRating() < minimum.getRating()) {
+                minimum = movie;
+            }
+            if (movie.getRating() > maximum.getRating()) {
+                maximum = movie;
+            }
+        }
+        System.out.println("The average rating of the movie is " + sum / movies.length);
+        System.out.println("The minimum rating of the movie is: " + minimum.getName() + " " + minimum.getRating());
+        System.out.println("The maximum rating of the movie is: " + maximum.getName() + " " + maximum.getRating());
+        System.out.println();
+    }
+
+    private int getMovieId(String name) {
+        for (int i = 0; i < movies.length; i++) {
+            if (movies[i].getName().equals(name) || movies[i].getName().equalsIgnoreCase(name)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private void searchMovie() {
+        System.out.println("Enter the movie name to search: ");
+        String name = new Scanner(System.in).nextLine();
+        int id = getMovieId(name);
+        if (id >= 0) {
+            System.out.println("Movie found: " + movies[id].getName() + " " + movies[id].getRating());
+            System.out.println();
+        } else {
+            System.out.println("Movie not found");
+        }
+        System.out.println();
+    }
+
+
+    private void updateRating() {
+        System.out.println("Enter the movie name to update: ");
+        String name = new Scanner(System.in).nextLine();
+        int id = getMovieId(name);
+        if (id >= 0) {
+            System.out.println("Enter the new movie rating: ");
+            double rating = new Scanner(System.in).nextDouble();
+            movies[id].setRating(rating);
+            System.out.println("Movie updated successfully!!");
+        } else {
+            System.out.println("Movie not found");
+        }
+        System.out.println();
+    }
+
+    private void deleteMovie() {
+        System.out.println("Enter the movie name to delete: ");
+        String name = new Scanner(System.in).nextLine();
+        int id = getMovieId(name);
+        if (id >= 0) {
+            movies[id] = null;
+            int length = movies.length;
+            Movie[] newMovies = new Movie[length - 1];
+            System.arraycopy(movies, 0, newMovies, 0, id);
+            System.arraycopy(movies, id + 1, newMovies, id, length - id - 1);
+            movies = newMovies;
+            System.out.println("Movie deleted successfully!");
+        } else {
+            System.out.println("Movie not found");
+        }
+        System.out.println();
+    }
+
+    private void sortMovie() {
+        double[] ratings = new double[movies.length];
+        for (int i = 0; i < movies.length; i++) {
+            ratings[i] = movies[i].getRating();
+        }
```

Comment:
instead of this logic you can send directly Movie object to the sorting, and use rating inside this method